stdlib/public/core/Filter.swift

```diff
@@ -304,20 +290,11 @@ extension LazySequenceProtocol {
   public func filter(
     _ isIncluded: @escaping (Elements.Element) -> Bool
   ) -> LazyFilterSequence<Self.Elements> {
-    return LazyFilterSequence(
-      _base: self.elements, isIncluded)
+    return LazyFilterSequence(_base: self.elements, isIncluded)
   }
 }
 
-% for Traversal in ['Forward', 'Bidirectional']:
-
-extension LazyCollectionProtocol
-%   if Traversal != 'Forward':
-  where
-  Self : ${collectionForTraversal(Traversal)},
-  Elements : ${collectionForTraversal(Traversal)}
-%   end
-{
+extension LazyCollectionProtocol {
```

Comment:
Will we still need a `LazyCollectionProtocol` now that everything can be handled conditionally?

Reply:
Oooh could we alias both `LazySequenceProtocol` and `LazyCollectionProtocol` to `Lazy`?

Reply:
I think it is possible to eventually get rid of `LazyCollectionProtocol` and instead do `extension LazySequenceProtocol where Base : Collection`, but I think it will be much easier to do once we conditionally conform all the lazy things.

Reply:
As for the name, `Lazy` sounds great but: I bet it is used in the wild with a different meaning, so this will be source breaking, and we also perhaps would like to one day have a "namespace" `Lazy`, so using it for a protocol is such a waste of a good identifier =)